File: 00. Estructura/01. Lenguajes de proposito general/02. Programación básica/06. Operadores/05. Bit a bit/MathLab/Bit a bit.m
     
     
% AND (Conjunción bit a bit de la representación binaria de las expresiones)
resultado = bitand(expresión1, expresión2);

% OR (Disjunción bit a bit de la representación binaria de las expresiones)
resultado = bitor(expresión1, expresión2);

% XOR (Desgualdad bit a bit de la representación binaria de las expresiones)
resultado = bitxor(expresión1, expresión2);

% NOT (Complemento bit a bit de la representación binaria de la expresión)
resultado = bitcmp(expresión);

% Desplazamiento a la izquierda (Desplazamiento bit a bit a la n-esima posición a la izquierda de la representación binaria de la expresión)
resultado = bitshift(expresión, n);

% Desplazamiento a la derecha (Desplazamiento bit a bit a la n-esima posición a la derecha de la representación binaria de la expresión)
resultado = bitshift(expresión, -n);

File: 00. Estructura/01. Lenguajes de proposito general/02. Programación básica/06. Operadores/05. Bit a bit/Java/Bit a bit.java

package prueba;

public class Prueba {

    public static void main(String[] args) {
       
        // AND (Conjunción bit a bit de la representación binaria de las expresiones)
        tipo resultado = expresión1 & expresión2;

        // OR (Disjunción bit a bit de la representación binaria de las expresiones)
        tipo resultado = expresión1 | expresión2;

        // XOR (Desgualdad bit a bit de la representación binaria de las expresiones)
        tipo resultado = expresión1 ^ expresión2;

        // NOT (Complemento bit a bit de la representación binaria de la expresión)
        tipo resultado = ~expresión;

        // Desplazamiento a la izquierda (Desplazamiento bit a bit a la n-esima posición a la izquierda de la representación binaria de la expresión)
        tipo resultado = expresión << n;

        // Desplazamiento a la derecha (Desplazamiento bit a bit a la n-esima posición a la derecha de la representación binaria de la expresión)
        tipo resultado = expresión >> n;

        // Desplazamiento a la derecha sin conservar el signo (Desplazamiento bit a bit a la n-esima posición a la derecha sin conservar el signo de la representación binaria de la expresión)
        tipo resultado = expresión >>> n;

    }
    
}
